duckduckgo/Android
# Feature/build process

Task/Issue URL: https://app.asana.com/0/search/828743867277634/817833722903104
Tech Design URL: 
CC: 

**Description**:
Changes to the project to support better build automation.

1. Moved the `version name` to its own file, called `version/version.properties`. This was needed to make it easier to grab during the build.
1. Moved the releases notes into source control, in `version/release-notes`. This just makes sense, generally, but was also needed to do more automated builds.
1. Added `fastlane` to the project, with a lane for uploading to GitHub releases. We wouldn't be likely to invoke this locally; this would be trigged from the release job.


**Steps to test this PR**:
1. **[LOCALLY]** Build an APK and then verify the `version name` and `version code` are what you expect based on the version set in `version.properties`. Hint, use `aapt dump badging path/to/app.apk | grep package` to easily get this info
1. **[LOCALLY]** Change the version in `version.properties` locally to something completely different and rebuild. Verify this works as expected, same as above
1. _Wait until our next release to see if it works_ 😰 


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
